packages/core-data/src/actions.js

```diff
@@ -342,6 +362,22 @@ export function* saveEntityRecord(
 		}
 	} catch ( _error ) {
 		error = _error;
+
+		// If we got to the point in the try block where we made an optimistic update,
+		// we need to roll it back here.
+		if ( persistedEntity && currentEdits ) {
+			yield receiveEntityRecords( kind, name, persistedEntity, undefined, true );
+			yield editEntityRecord(
+				kind,
+				name,
+				recordId,
+				{
+					...currentEdits,
+					...( yield select( 'getEntityRecordEdits', kind, name, recordId ) ),
```

Comment:
Do we really need this though? Isn't it the regular behavior of `editEntityRecord` to merge?

Reply:
Yes, but the old values would overwrite the new ones made in-flight.

Reply:
Right 👍